src/executionFiles/bridges/nxtp.execute.ts

```diff
@@ -201,22 +215,38 @@ export class NXTPExecutionManager {
       return status
     }
 
-    const fulfillPromise = nxtpSDK.fulfillTransfer(preparedTransaction)
-
     // STEP 7: Wait for signature //////////////////////////////////////////////////////////
-    await nxtpSDK.waitFor(
-      NxtpSdkEvents.ReceiverPrepareSigned,
-      20 * 60 * 1000, // = 20 minutes
-      (data) => data.transactionId === transactionId
+    const calculatedRelayerFee = await this.calculateRelayerFee(
+      nxtpSDK,
+      preparedTransaction
+    )
+
+    const signature = await signFulfillTransactionPayload(
+      preparedTransaction.txData.transactionId,
+      calculatedRelayerFee,
+      preparedTransaction.txData.receivingChainId,
+      preparedTransaction.txData.receivingChainTxManagerAddress,
+      signer
     )
 
     claimProcess.status = 'PENDING'
     claimProcess.message = 'Waiting for claim'
     update(status)
 
+    const callData = await this.decryptData(
```

Comment:
It is great that we are now in charge of handling the decryption. We should update the claimProcess, if the user has to decrypt the data by setting the message and status right.

Reply:
if no decrypt hook was defined by the user we don't need to update the status.